rust/core-lib/src/editor.rs

```diff
@@ -357,8 +359,174 @@ impl Editor {
+                                } else {
+                                    state = State::Finished
+                                }
+                            }
+                            State::EvenNumberedRIS => {
+                                if is_regional_indicator_symbol(code_point) {
+                                    delete_code_point_count -= 1;
+                                    state = State::OddNumberedRIS;
+                                } else {
+                                    state = State::Finished;
+                                }
+                            }
+                            State::BeforeKeycap => {
+                                if is_variation_selector(code_point) {
+                                    last_seen_vs_code_point_count = 1;
+                                    state = State::BeforeVsAndKeycap;
+                                } else {
+                                    if is_keycap_base(code_point) {
+                                        delete_code_point_count += 1;
+                                    }
+                                    state = State::Finished;
+                                }
+                            }
+                            State::BeforeVsAndKeycap => {
+                                if is_keycap_base(code_point) {
+                                    delete_code_point_count += last_seen_vs_code_point_count + 1;
+                                }
+                                state = State::Finished;
+                            }
+                            State::BeforeEmojiModifier => {
+                                if is_variation_selector(code_point) {
+                                    last_seen_vs_code_point_count = 1;
+                                    state = State::BeforeVSAndEmojiModifier;
+                                } else {
+                                    if is_emoji_modifier_base(code_point) {
+                                        delete_code_point_count += 1;
+                                    }
+                                    state = State::Finished;
+                                }
+                            }
+                            State::BeforeVSAndEmojiModifier => {
+                                if is_emoji_modifier_base(code_point) {
+                                    delete_code_point_count += last_seen_vs_code_point_count + 1;
+                                }
+                                state = State::Finished;
+                            }
+                            State::BeforeVS => {
+                                if is_emoji(code_point) {
+                                    delete_code_point_count += 1;
+                                    state = State::BeforeEmoji;
+                                } else {
+                                    if !is_variation_selector(code_point) { //TODO: UCharacter.getCombiningClass(codePoint) == 0
```

Comment:
Is this TODO for future work or for now?

Reply:
That's from Android's [getOffsetForBackspaceKey:249](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/text/method/BaseKeyListener.java#249). I need to write a function to get combining class (maybe in future work?) . However, without this, I passed all tests provided by Android :laughing: .